src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -14,7 +14,22 @@ namespace Stratis.Bitcoin.Consensus
     /// </summary>
     public interface IChainedHeaderValidator
     {
-        void Validate(ChainedHeader chainedHeader);
+        /// <summary>
+        /// Validation of a header that was seen for the first time.
+        /// </summary>
+        /// <param name="chainedHeader">The chained header to be validated.</param>
+        void ValidateHeader(ChainedHeader chainedHeader);
+
+        /// <summary>
+        /// Validate the block signature (for POS) and merkle tree, throws if validation fails. 
```

Comment:
Maybe put this into the remarks as it is implementation specific thing and that the describe it on higher level.

Verifies that the block data corresponds to the chain header.

Reply:
In fact that level of description should go in the implementation itself `(for POS) and merkle tree,`

Reply:
yes